ci/travis/get_build_info.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+"""
+This script gathers build metadata from Travis environment variables and Travis
+APIs.
+
+Usage:
+$ python get_build_info.py
+{
+    "json": ["containing", "build", "metadata"]
+}
+"""
+
+import os
+import json
+
+import requests
+
+
+def get_build_env():
+    keys = [
+        "TRAVIS_BRANCH", "TRAVIS_BUILD_ID", "TRAVIS_BUILD_NUMBER",
+        "TRAVIS_BUILD_WEB_URL", "TRAVIS_COMMIT", "TRAVIS_COMMIT_MESSAGE",
+        "TRAVIS_DIST", "TRAVIS_JOB_ID", "TRAVIS_JOB_NUMBER",
+        "TRAVIS_JOB_WEB_URL", "TRAVIS_OS_NAME"
+    ]
+    return {key: os.environ.get(key) for key in keys}
+
+
+def get_build_config():
+    url = "https://api.travis-ci.com/job/{job_id}?include=job.config"
+    url = url.format(job_id=os.environ["TRAVIS_JOB_ID"])
+    resp = requests.get(url, headers={"Travis-API-Version": "3"})
```

Comment:
Not terribly important but my suggestion would be to use `urllib` here. (You can even just copy-paste `def download(url)` from `setup.py`.)

`requests` introduces a dependency (which seems overkill for this?) and I've seen `requests` fails on some systems (older Macs? I forget) due to SSL issues; I struggled with this a lot when trying to get `setup.py` implemented and working on all our systems a few weeks ago.

Reply:
good point. I was assuming that by the time this script is executed, we had the CI env setup and ran all the tests with it.